cpp/src/icegriddb/IceGridDB.cpp

```diff
@@ -301,9 +275,17 @@ run(const Ice::StringSeq& args)
 
             if(!serverVersion.empty())
             {
-                Ice::ValueFactoryPtr factory = new ValueFactoryI(serverVersion);
-                communicator->getValueFactoryManager()->add(factory, "::IceGrid::ServerDescriptor");
-                communicator->getValueFactoryManager()->add(factory, "::IceGrid::IceBoxDescriptor");
+                communicator->getValueFactoryManager()->add([serverVersion](const string& type)
```

Comment:
This fails to build in release mode because type is unused
```
D:\4.0\icegrid\cpp\src\icegriddb\IceGridDB.cpp(282,21): warning C4100: 'type': unreferenced formal parameter [D:\4.0\icegrid\cpp\src\icegriddb\msbuild\icegriddb.vcxproj]
         D:\4.0\icegrid\cpp\src\icegriddb\IceGridDB.cpp(288,21): warning C4100: 'type': unreferenced formal parameter [D:\4.0\icegrid\cpp\src\icegriddb\msbuild\icegriddb.vcxproj]
```

Reply:
I'll just remove the assert I guess.